### PR TITLE
Added support for Django unittests 

### PR DESCRIPTION
**Pull request by mh-firouzjah**
Update `unittestScripts.ts` will help the extension to recognize the Django tests and run them with `DJANGO_SETTINGS_MODULE` set and also by calling `django.setup()` it will make sure that all django apps are load and it's ready to go for tests.

### CODE REVIEW

File: src/unittest/unittestScripts.ts [hunk 1]
```diff
@@ -12,6 +12,22 @@ import base64
 import json
 import traceback
 
+from pathlib import Path
+start_dir = sys.argv[2]
```
Comment (kondratyev-nv) [inline src/unittest/unittestScripts.ts:16]: This is already defined below, can you please reuse this?

Comment (kondratyev-nv) [inline src/unittest/unittestScripts.ts:16]: This extension still supports Python 2.7 and pathlib was added in 3.4. Can you please either make it compatible or somehow skip Django for 2.7? It shouldn't fail existing tests (see CI).

File: src/unittest/unittestScripts.ts [hunk 2]
```diff
@@ -12,6 +12,22 @@ import base64
...
+if Path(start_dir + "/manage.py").is_file():
```
Comment (kondratyev-nv) [inline src/unittest/unittestScripts.ts:17]: I might be wrong, but you might need to use `os.pathsep` here.

File: src/unittest/unittestScripts.ts [hunk 3]
```diff
@@ -12,6 +12,22 @@ import base64
...
+    with open(start_dir + "/manage.py", "r") as management_file:
+        contents = management_file.readlines()
+        if any(True for line in contents if line.strip().replace('"""', '') == "Django\'s command-line utility for administrative tasks."):
```
Comment (kondratyev-nv) [inline src/unittest/unittestScripts.ts:20]: Is this only to check that this is a django app?

File: src/unittest/unittestScripts.ts [hunk 4]
```diff
@@ -12,6 +12,22 @@ import base64
 import json
 import traceback
 
+from pathlib import Path
```
Comment (kondratyev-nv) [inline src/unittest/unittestScripts.ts:15]: Can you please wrap all this logic to a function and call it below before the main logic is triggered?

**Review by kondratyev-nv — CHANGES_REQUESTED** [review]
@mh-firouzjah Thanks for the PR! Just to let you know, this extension has its own logic and not directly related to the [Python extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python).
Couple of other things:
1. Please, add tests to make sure that this is working.
2. It seems to be failing for Python 2.7, see CI.

There seems to be an issues with Python 3.6 CI, so you can ignore this for now.

File: src/unittest/unittestScripts.ts (hunk 1 above)
Comment (mh-firouzjah) [reply src/unittest/unittestScripts.ts:16]: use, it is possible to reuse it.

Comment (mh-firouzjah) [reply src/unittest/unittestScripts.ts:16]: I'll change the code to fix this

File: src/unittest/unittestScripts.ts (hunk 3 above)
Comment (mh-firouzjah) [reply src/unittest/unittestScripts.ts:20]: the code at line 20? yes I want to make sure the file is generated by django-admin startproject

File: src/unittest/unittestScripts.ts (hunk 4 above)
Comment (mh-firouzjah) [reply src/unittest/unittestScripts.ts:15]: wilco

**Comment by mh-firouzjah** [comment]
the idea is to check if an auto-generated `manage.py` exists in the `rootfolder` and by auto-generated I mean it is generated by `django-admin startproject` command.
existence of such a file has two messages, first this is a django project and not a pure python project so tests are expected to be django unittests and second, the content of this file can be used to figure out where is the path of the django project settings and by use of `eval` it will automatically set the `DJANGO_SETTINGS_MODULE` env variable.
also I'm not sure it's enough or not, but I've added an extra line to use 'ast.literal_eval` and check if the value that is going to be set for `DJANGO_SETTINGS_MODULE` is a safe python expression.

**Comment by mh-firouzjah** [comment]
a new PR has been sent.

**Pull request closed without merging**